content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,563 @@
+---
+title: Train a Recurrent Neural Network (RNN) with Poutyne
+author: David Beauchemin
+date: '2020-08-18'
+slug: machine learning
+type: post
+categories: ["python", "machine learning"]
+tags: []
+description: "Address tagging with a RNN"
+featured: "OpenLayer_YoutubeBanner.jpg"
+featuredpath: "img/headers/"
+---
+
+> In this article, we will train an RNN, or more precisely, an LSTM, to predict the sequence of tags associated with a 
+given address, known as parsing address. 
+
+> Also, the code (only) is available in [this Google Colab Jupyter notebook](https://colab.research.google.com/github/dot-layer/blog/blob/post%2Fdb_sequence_training_poutyne/content/blog/2020-08-19-train-a-sequence-model-with-poutyne/article_notebook.ipynb#scrollTo=c1JxpCRsQN0e).
+
+> Before starting this article, I would like to disclaim that this tutorial is greatly inspired by a online tutorial I 
+created for Poutyne framework. Also, the content is based on a recent article I wrote with Marouane Yassine. However, 
+there are differences between the present work and the two others, I've tried to put more insights for the less technical reader in this one.
+>
+Sequential data, such as addresses, are pieces of information that are deliberately given in a specific order. In other words, they are sequences with a particular structure; and knowing this structure is crucial for predicting the missing entries of a given truncated sequence. For example, 
+when writing an address, we know, in Canada, that after the civic number (e.g. 420), we have the street name (e.g. du Lac).
+Hence, if one was asked to complete an address containing only a number, she could reasonably assume that the next information that should be added to the sequence is a street name. Various modelling approaches have been proposed to make predictions over sequential data. Still, more recently, deep learning models known as Recurrent Neural Network (RNN) has been introduced for this type of data.
+
+Training an RNN requires various tricks (padding and packing) that we will explore in this article. First, let's 
+state our problem, and later on, we will discuss what an actual RNN or LSTM is.
+
+## Address Tagging
+Address tagging is the task of detecting, by tagging, the different parts of an address such as the civic number, 
+the street name or the postal code (or zip code). The following figure shows an example of such a tagging.
+
+![address parsing canada](address_parsing.png)
+
+Since addresses are written in a predetermined sequence, RNN is the best way to crack this problem. But to decode the output of the RNN, we also need another component, a fully-connected layer. 
```

Comment:
```suggestion
Since addresses are written in a predetermined sequence, RNN is the best way to crack this problem. However, to decode the output of the RNN, we also need another component, a fully-connected layer. 
```